contracts/StakingPool.sol

```diff
@@ -1,43 +1,99 @@
-pragma solidity ^0.5.0;
-
+/* solium-disable security/no-block-members */
+/* solium-disable security/no-inline-assembly */
+pragma solidity >=0.5.0 <0.6.0;
 import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
 import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
 import "openzeppelin-solidity/contracts/token/ERC20/ERC20Detailed.sol";
 import "openzeppelin-solidity/contracts/token/ERC20/ERC20Burnable.sol";
 import "./math.sol";
 import "./token/ApproveAndCallFallBack.sol";
+import "./token/MiniMeTokenInterface.sol";
 
 contract StakingPool is ERC20, ERC20Detailed, ERC20Burnable, DSMath, ApproveAndCallFallBack {
-  uint private INITIAL_SUPPLY = 0;
-  IERC20 public token;
+  uint public MAX_SUPPLY = 0;
+
+  MiniMeTokenInterface public token;
+  uint public stakingBlockLimit;
+  uint public blockToCheckBalance;
 
-  constructor (address tokenAddress) public ERC20Detailed("TellerStatus", "TSNT", 18) {
-    token = IERC20(tokenAddress);
+  /**
+   * @param _tokenAddress SNT token address
+   * @param _stakingPeriodLen Number of block that represents the period when Staking is available
+   */
+  constructor (address _tokenAddress, uint _stakingPeriodLen) public ERC20Detailed("TellerStatus", "TSNT", 18) {
```

Comment:
Do we still want to call the token TellerStatus? I think now it's going to be more generic no?

Reply:
You're right. 
Any proposals?    "SDT" -> "Status DAO Token"? 
cc: @iurimatias

Reply:
That's a good name IMO (though a bit similar to STD lol)